### ISSUE TITLE
`aria-hidden` remains after closing popover with nested dropdown

### ISSUE THREAD

**Issue by qroll** [LifeSG/react-design-system#913]
**Component**
`Popover` + `InputSelect`

**Describe the bug**
If you have a dropdown-based component in a popover, `aria-hidden` remains on the main page elements after closing both the dropdown + popover at the same time

This causes unit tests or e2e tests to fail as the locators typically check that the element is accessible

**To Reproduce**

```
<PopoverInline
    content="fox"
    popoverContent={
        <div>
            hello <InputSelect options={[]} />
        </div>
    }
/>
```

1. Activate popover
2. Activate select dropdown
3. Click outside of popover
4. Observe that `aria-hidden="true"` remains on the parent container

**Expected behavior**
`aria-hidden` should have been cleaned up when all floating element are closed

**Additional context**
Changing to `modal={false}` in `ElementWithDropdown` resolves this, so it may be some weird interaction with nested modal + non-modal floating elements
